### DISCUSSION TITLE
MonitoredItems filters

### DISCUSSION THREAD

**Discussion by dremsol** [FreeOpcUa/opcua-asyncio#1280] [Q&A] [answered]
When subscribing to a list of Nodes when using `deadband_monitor` a `DataChangeFilter` is tied to the subscription and used for each monitoreditem in the list. However, when having a different filter for the Nodes in the list how should we use this function?

According to the [MonitoredItem model](https://reference.opcfoundation.org/Core/Part4/v104/docs/5.12.1) each time a [MonitoredItem](https://reference.opcfoundation.org/search/17?t=MonitoredItem) is sampled, the [Server](https://reference.opcfoundation.org/search/17?t=Server) evaluates the sample using the filter defined for the [MonitoredItem](https://reference.opcfoundation.org/search/17?t=MonitoredItem). 

Something I fail to understand is when creating separate subscriptions for nodes with different DataChangeFilters, returning different subscription handles, are still visible under the same subscriptionid in my S71500. 

Could someone clarify how this works?

**Comment by schroeder-** [comment, marked as answer]
You can create different monitored items with multiple calss `deadband_monitor` or `subscribe_data_change` or even `_make_monitored_item_request`. 
I am not clear what you mean. So are you using multiple `create_subscription` or a you using on subscription with different Monitored items with differenent filters. 
For each create_subscription a "SubscriptionId" is created. Each SubscriptionId can create multiple MonitoredItems, which all get a own "MonitoredItemId". Hope it helps.
